gradio/__init__.py

```diff
@@ -240,3 +241,27 @@
     "update",
     "DeepLinkButton",
 ]
+
+
+def i18n(**kwargs) -> _I18n:
+    """Create an i18n instance with translations. Use this function to define
+    translation dictionaries for different locales.
+
+    Args:
+        **kwargs: Each keyword argument should be a locale code (e.g., "en", "fr") with a
+        dictionary value, which maps translation keys to translated strings.
+
+    Returns:
+        An I18n instance with the specified translations.
+
+    Example:
+        i18n = gr.i18n(
+            en={"greeting": "Hello"},
+            es={"greeting": "Hola"}
+        )
+        with gr.Blocks() as demo:
+            gr.Textbox(label=i18n("greeting"))
+
+        demo.launch(i18n=i18n)
+    """
+    return _I18n(translations=kwargs)
```

Comment:
This approach of using a factory method to create an internal class works, but a more standard approach would be to just have `i18n` as a class that a user instantiates directly, like this:
```
class i18n:
    def __init__(self, **translations: dict[str, str]):
```

Benefit of this, besides being simpler, is that we'll want to update all of the typing for all the parameters for all of components that can also accept i18n objects, and we can reference this class directly. So for example, in `Textbox`, the parameter

```
label: str | None = None,
```

should now be:

```
label: str | i18n | None = None,
```